examples/loaders/resourceprovider-dummy-fs.mjs

```diff
@@ -0,0 +1,31 @@
+import * as fs from 'fs';
+
+export let prefixes = ['file:'];
+
+export let suffixes = [];
+
+// Following was adapted from the example in the Node.js docs
+// https://nodejs.org/api/esm.html#esm_code_resolve_code_hook
+export async function resolve(specifier, context, defaultResolve) {
+  console.log(
+    `### Resolving resource in dummy resourceProvider for ${specifier}\n`
+  );
+
+  return defaultResolve(specifier, context, defaultResolve);
+}
+
+export function getResourceProvider() {
+  return {
+    async getResource(url) {
+      console.log(
+        `### Getting resource in dummy resourceProvider for ${url}\n`
+      );
+
+      return new Promise((resolve, reject) => {
```

Comment:
Tip: You can use `fs.promises`. E.g.:

```
import {promises as fs} from 'fs';

fs.readFile(new URL(url), 'utf8'); // returns a promise!
```

Reply:
Awesome tip, I've incorporated this now, thanks!